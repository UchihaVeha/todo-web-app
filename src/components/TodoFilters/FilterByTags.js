// @flow
import * as React from 'react';
import { withStyles } from 'material-ui/styles';
import { MenuItem } from 'material-ui/Menu';
import { InputLabel } from 'material-ui/Input';
import { FormControl } from 'material-ui/Form';
import { Select } from 'material-ui';
import type { TagIds, Tags } from 'modules/tag';

const styles = theme => ({
  select: {
    width: '100% !important',
    '& button': {
      fill: `${theme.palette.common.darkWhite} !important`
    },
    '& hr': {
      borderBottom: `1px solid ${theme.palette.common.darkWhite} !important`
    },
    '& div': {
      color: `${theme.palette.common.darkWhite} !important`
    }
  },
  formControl: {
    width: '100%'
  }
});

type Props = {
  tags: Tags,
  value: TagIds,
  onChange: (tags: number[]) => void,
  classes: Return<typeof styles>
};
class FilterByTags extends React.Component<Props> {
  handleChange = (e: SyntheticEvent<{ value: any[] }>): void => {
    this.props.onChange(e.target.value);
  };

  render() {
    const { classes, value, tags } = this.props;
    return (
      <FormControl className={classes.formControl}>
        <InputLabel htmlFor="age-simple">Tags</InputLabel>
        <Select
          multiple
          displayEmpty
          value={value.toArray()}
          onChange={this.handleChange}
          className={classes.select}
        >
          {tags
            .map(tag => (
              <MenuItem key={tag.id} value={tag.id}>
                {tag.name}
              </MenuItem>
            ))
            .toArray()}
        </Select>
      </FormControl>
    );
  }
}

export default withStyles(styles, { name: 'FilterByTags' })(FilterByTags);
