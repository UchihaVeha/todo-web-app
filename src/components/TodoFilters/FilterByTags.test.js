import React from 'react';
import { shallow } from 'enzyme';
import { List } from 'immutable';
import FilterByTags from './FilterByTags';

describe('TodoFilters/FilterByTags component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<FilterByTags tags={List([])} value={List()} />);
  });

  it('should render withStyles', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component', () => {
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
