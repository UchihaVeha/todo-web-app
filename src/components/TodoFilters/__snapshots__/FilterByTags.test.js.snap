// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoFilters/FilterByTags component should render component 1`] = `
<withStyles(FormControl)
  className="FilterByTags-formControl-2"
>
  <withStyles(InputLabel)
    htmlFor="age-simple"
  >
    Tags
  </withStyles(InputLabel)>
  <withStyles(Select)
    className="FilterByTags-select-1"
    displayEmpty={true}
    multiple={true}
    onChange={[Function]}
    value={Array []}
  />
</withStyles(FormControl)>
`;

exports[`TodoFilters/FilterByTags component should render withStyles 1`] = `
<FilterByTags
  classes={
    Object {
      ".FilterByTags-select-1 button": "FilterByTags-select-1 button",
      ".FilterByTags-select-1 div": "FilterByTags-select-1 div",
      ".FilterByTags-select-1 hr": "FilterByTags-select-1 hr",
      "formControl": "FilterByTags-formControl-2",
      "select": "FilterByTags-select-1",
    }
  }
  tags={Immutable.List []}
  value={Immutable.List []}
/>
`;
